install_prerequisites.py

```diff
@@ -117,14 +121,20 @@ def main():
         # get prerequisites
         cmd = ['sudo','yum','install','-y','epel-release','wget','openssl','ca-certificates']
         build.run_cmd(cmd, check_rc='installing epel failed')
-        cmd = ['sudo','yum','install','-y','curl','gcc-c++','git','autoconf','automake','texinfo',
+        package_list = ['curl','gcc-c++','git','autoconf','automake','texinfo',
                'help2man','rpm-build','rubygems','ruby-devel','zlib-devel','fuse','fuse-devel',
                'bzip2-devel','libcurl-devel','libmicrohttpd-devel','libxml2-devel','libtool','libuuid-devel','openssl-devel','unixODBC-devel','patchelf']
-        if distro_id in ['rocky', 'almalinux']:
-            cmd.append('python36-devel') # python39-devel also available.
+
+        # For some versions, curl is installed by another step of this process and manually installing the package here
+        # creates a conflict. Rather than conditionally installing curl (or any other package which may cause problems
+        # in the future) we use --skip-broken to avoid broken packages.
+        cmd = ['sudo','yum','install','-y','--skip-broken']
```

Comment:
This only applies to centos, right? Has this always been the case?

I'm a little confused because `curl` is in the list above. How does `--skip-broken` play into this?

Reply:
The package conflict occurs on Rocky 9. `--skip-broken` skips installing packages which would be broken if installed. It is harmless if not needed, so we always specify it here, but we can make it conditional. It's not clear to me that this would apply to all platforms which is why I did it this way

Reply:
it feels a bit loose/undefined over time, but if it's the best we have, seems okay.

Reply:
I'm thinking we make it conditional. At the very least, that provides signal that Rocky needed some special attention.

Also, Rocky supports `yum`? Shouldn't we be using `dnf` instead?

Reply:
> provides signal

yes, exactly.

Reply:
Not sure. We are mixing `yum` and `dnf` above as well.

I am also not convinced this is Rocky specific. It may also affect Almalinux and other EL distros.

That being said, I will look into rewriting this.

Reply:
Please review based on latest changes. Resolving